Docs: Adds best practices after visit and a link back to e2e.md (#21117)
grafana/grafana: contribute/style-guides/e2e.md

The last step in our example is to use our `Login` page as part of a test. The `pageFactory` function we used before gives us two things: 
- The `url` property is used whenever we call the `visit` function and is equivalent to the Cypress function [cy.visit()](https://docs.cypress.io/api/commands/visit.html#Syntax).

> > Best practices after calling visit is to always call `should` on a selector to prevent flaky tests where you try to operate on a selector that isn't ready. Read more about it [here](https://docs.cypress.io/guides/core-concepts/retry-ability.html#Commands-vs-assertions).  

Reviewer: ```suggestion
> Best practice after calling `visit` is to always call `should` on a selector to prevent flaky tests when you try to access an element that isn't ready. Read more about it [here](https://docs.cypress.io/guides/core-concepts/retry-ability.html#Commands-vs-assertions).  
```

- Any defined selector in the `selectors` property can be accessed from the `Login` page by invoking it and this is equivalent to the result of the Cypress function [cy.get(...)](https://docs.cypress.io/api/commands/get.html#Syntax).
```ecmascript 6
describe('Login test', () => {
  it('Should pass', () => {
    e2e.pages.Login.visit();

>     e2e.pages.Login.username().should('be.visible'); // To prevent flaky tests always do a .should on an selector that you expect to be in the DOM. Read more here: https://docs.cypress.io/guides/core-concepts/retry-ability.html#Commands-vs-assertions

Reviewer: Maybe add that as a block comment above this line for readability

Author: Good suggestion!

The last step in our example is to use our `Login` page as part of a test. The `pageFactory` function we used before gives us two things: 
- The `url` property is used whenever we call the `visit` function and is equivalent to the Cypress function [cy.visit()](https://docs.cypress.io/api/commands/visit.html#Syntax).

> > Best practice after calling `visit` is to always call `should` on a selector to prevent flaky tests when you try to access an element that isn't ready. Read more about it [here](https://docs.cypress.io/guides/core-concepts/retry-ability.html#Commands-vs-assertions).  

Reviewer 2: ```suggestion
> Best practice after calling `visit` is to always call `should` on a selector to prevent flaky tests when you try to access an element that isn't ready. For more information, refer to [Commands vs. assertions](https://docs.cypress.io/guides/core-concepts/retry-ability.html#Commands-vs-assertions).  
```

> - Any defined selector in the `selectors` property can be accessed from the `Login` page by invoking it and this is equivalent to the result of the Cypress function [cy.get(...)](https://docs.cypress.io/api/commands/get.html#Syntax).

Reviewer 2: ```suggestion
- Any defined selector in the `selectors` property can be accessed from the `Login` page by invoking it. This is equivalent to the result of the Cypress function [cy.get(...)](https://docs.cypress.io/api/commands/get.html#Syntax).
```

```ecmascript 6
describe('Login test', () => {
  it('Should pass', () => {
    e2e.pages.Login.visit();

>     // To prevent flaky tests always do a .should on an selector that you expect to be in the DOM. 

Reviewer 2: ```suggestion
    // To prevent flaky tests, always do a .should on any selector that you expect to be in the DOM. 
```

```

Now we can go ahead and write our test and the one thing that differs from the `Basic example` is that we pass in which data source we want to click on as an argument to the selector function:

> > Best practices after calling visit is to always call `should` on a selector to prevent flaky tests where you try to operate on a selector that isn't ready. Read more about it [here](https://docs.cypress.io/guides/core-concepts/retry-ability.html#Commands-vs-assertions).  

Reviewer: See my comment above

```

> Now we can go ahead and write our test and the one thing that differs from the `Basic example` is that we pass in which data source we want to click on as an argument to the selector function:

Reviewer 2: ```suggestion
Now we can write our test. The one thing that differs from the `Basic example` is that we pass in which data source we want to click on as an argument to the selector function:
```
